src/serde/deserialize_program.rs

```diff
@@ -31,16 +35,38 @@ pub struct FlowTrackingData {
     #[serde(deserialize_with = "deserialize_map_to_string_and_bigint_hashmap")]
     pub reference_ids: HashMap<String, BigInt>,
 }
+
 #[derive(Deserialize, Debug, Clone, PartialEq)]
 pub struct ApTracking {
     pub group: usize,
     pub offset: usize,
 }
+
 #[derive(Deserialize, Debug)]
 pub struct Identifier {
     pub pc: Option<usize>,
 }
 
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct ReferenceManager {
+    pub references: Vec<Reference>,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct Reference {
+    pub ap_tracking_data: ApTracking,
+    pub pc: Option<usize>,
+    #[serde(deserialize_with = "deserialize_value_address")]
+    #[serde(rename(deserialize = "value"))]
+    pub value_address: ValueAddress,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct ValueAddress {
+    pub register: Option<Register>,
+    pub offset: Option<i32>,
+}
```

Comment:
In which scenario would we have a `ValueAddress` with no associated register?
Related, can't we just always treat the offset as `i32`? An offset of `0` is the same as no offset.

Reply:
Also, is the offset guaranteed to fit in 32 bits?

Reply:
There are more references in the `reference manager` than just the ones used by hints. For the moment we don´t need them, so those references are set with None values in the `ValueAddress`.
If we wanted them, there should be another logic for parsing those references. They have to be present though,  because references are accessed by an id (`reference_ids` of `FlowTrackingData` struct), and that id corresponds to the index of the list of references in the reference manager.

Reply:
If this sounds a little confusing let me know and I will explain in more detail :)